dotnet/src/Agents/Core/Chat/KernelFunctionTerminationStrategy.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel.Agents.Chat;
+
+/// <summary>
+/// Signals termination based on the evaluation of a <see cref="KernelFunction"/>.
+/// </summary>
+/// <param name="function">A <see cref="KernelFunction"/> used for termination criteria</param>
+public class KernelFunctionTerminationStrategy(KernelFunction function) : TerminationStrategy
+{
+    /// <summary>
+    /// A well-known <see cref="KernelArguments"/> key associated with the agent name.
+    /// </summary>
+    public const string ArgumentKeyAgent = "_agent_";
+
+    /// <summary>
+    /// A well-known <see cref="KernelArguments"/> key associated with the chat history.
+    /// </summary>
+    public const string ArgumentKeyHistory = "_history_";
+
+    /// <summary>
+    /// Optional arguments used when invoking <see cref="KernelFunctionTerminationStrategy.Function"/>.
+    /// </summary>
+    public KernelArguments? Arguments { get; init; }
+
+    /// <summary>
+    /// The <see cref="Microsoft.SemanticKernel.Kernel"/> used when invoking <see cref="KernelFunctionTerminationStrategy.Function"/>.
+    /// </summary>
+    public Kernel Kernel { get; init; } = Kernel.CreateBuilder().Build();
```

Comment:
nit: new Kernel()?

Reply:
I think I can update a couple other spots also.  For some reason I missed the simple constructor when looking.  Should've just tried it.

Reply:
Updated